crates/hyperswitch_connectors/src/connectors/razorpay.rs

```diff
@@ -204,12 +216,93 @@ impl ConnectorIntegration<SetupMandate, SetupMandateRequestData, PaymentsRespons
 {
 }
 
+impl api::PaymentsCreateOrder for Razorpay {}
+
+impl ConnectorIntegration<CreateOrder, CreateOrderRequestData, PaymentsResponseData> for Razorpay {
+    fn get_headers(
+        &self,
+        req: &CreateOrderRouterData,
+        connectors: &Connectors,
+    ) -> CustomResult<Vec<(String, Maskable<String>)>, errors::ConnectorError> {
+        self.build_headers(req, connectors)
+    }
+
+    fn get_content_type(&self) -> &'static str {
+        self.common_get_content_type()
+    }
+
+    fn get_url(
+        &self,
+        _req: &CreateOrderRouterData,
+        connectors: &Connectors,
+    ) -> CustomResult<String, errors::ConnectorError> {
+        Ok(format!("{}v1/orders", self.base_url(connectors)))
+    }
+
+    fn get_request_body(
+        &self,
+        req: &CreateOrderRouterData,
+        _connectors: &Connectors,
+    ) -> CustomResult<RequestContent, errors::ConnectorError> {
+        let minor_amount = req.request.minor_amount;
+        let currency = req.request.currency;
+        let amount = convert_amount(self.amount_converter, minor_amount, currency)?;
+        let connector_router_data = razorpay::RazorpayRouterData::try_from((amount, req))?;
+        let connector_req = razorpay::RazorpayOrderRequest::try_from(&connector_router_data)?;
+        Ok(RequestContent::Json(Box::new(connector_req)))
+    }
+
+    fn build_request(
+        &self,
+        req: &CreateOrderRouterData,
+        connectors: &Connectors,
+    ) -> CustomResult<Option<Request>, errors::ConnectorError> {
+        let req = Some(
+            RequestBuilder::new()
+                .method(Method::Post)
+                .attach_default_headers()
+                .headers(CreateOrderType::get_headers(self, req, connectors)?)
+                .url(&CreateOrderType::get_url(self, req, connectors)?)
+                .set_body(CreateOrderType::get_request_body(self, req, connectors)?)
+                .build(),
+        );
+        Ok(req)
+    }
+
+    fn handle_response(
+        &self,
+        data: &CreateOrderRouterData,
+        event_builder: Option<&mut ConnectorEvent>,
+        res: Response,
+    ) -> CustomResult<CreateOrderRouterData, errors::ConnectorError> {
+        let response: razorpay::RazorpayOrderResponse = res
+            .response
+            .parse_struct("razorpay RazorpayOrderResponse")
```

Comment:
Nit: Probably don't need to mention RazorPay twice.

```suggestion
            .parse_struct("RazorpayOrderResponse")
```